rand_seeder/src/sip.rs

```diff
@@ -106,6 +120,60 @@ impl SipHasher {
             hasher: Hasher::new_with_keys(key0, key1)
         }
     }
+    
+    /// Finish writes and convert the core into an RNG
+    pub fn make_rng(self) -> SipRng {
+        self.hasher.make_rng()
+    }
+}
+
+impl SipRng {
+    // private constructor
+    fn from_state(state: State) -> SipRng {
+        SipRng {
+            state,
+            adj: 0x13,
+        }
+    }
+}
+
+impl RngCore for SipRng {
+    fn next_u32(&mut self) -> u32 {
+        // Lazy way to implement. Good enough for seeding RNGs.
+        self.next_u64() as u32
+    }
+    
+    fn next_u64(&mut self) -> u64 {
+        self.state.v2 ^= self.adj;
+        self.adj = self.adj.wrapping_sub(0x11);
+        
+        Sip24Rounds::c_rounds(&mut self.state);
+        
+        self.state.v0 ^ self.state.v1 ^ self.state.v2 ^ self.state.v3
```

Comment:
Your comment above refers to this code, right? What is the motivation for this construction?

Reply:
XORing the state together? No, I guess you mean using two rounds here?

Well, the default construction of SIP hash uses four rounds after consuming all input (although this is configurable, and the authors invite others to try attacking the hash function with different constructions, even with zero rounds at the end).

Using two rounds here, then two rounds before each generated output maintains four rounds between input and output phases, with two rounds between each output mirroring the input.

I'm not qualified to say anything about the security, but it mirrors the input design and seems a reasonable construction. I was wondering about asking the SIP hash authors for their input on this.

Reply:
I should add that all the variants of SIP hash mutate the state somehow between each set of rounds, in addition to mutation by input. Mutating the adjustor (`adj`) is my invention to add unique input each time; as I understand the bit propagation of the "rounds" is perhaps good enough alone, but I believe this increases the strength vs any kind of analysis.

Of course, making SIP rng stronger vs crytoanalysis is beyond our requirements, but still seems prudent to do what's simple.